optimizer/optimizer.go

```diff
@@ -61,13 +54,30 @@ func Optimize(is infoschema.InfoSchema, ctx context.Context, node ast.Node) (pla
 	return bestPlan, nil
 }
 
+// Prepare prepares a raw statement parsed from parser.
+// The statement must be prepared before it can be passed to optimize function
+// We pass InfoSchema instead of get from Context in case it is changed after resolving name.
+func Prepare(is infoschema.InfoSchema, ctx context.Context, node ast.Node) error {
+	if err := Validate(node, true); err != nil {
```

Comment:
`Compile` is used for non-prepared statement, the argument in for `Validate` is `false`, in `Prepare`,  the argument for `Validate` is `true`.